### PR TITLE
Avoid exposing PrimaryWorkspace to LUT/SBD

### PR DESCRIPTION
**Pull request by tmat**
Required for https://github.com/dotnet/roslyn/issues/45730 and https://github.com/dotnet/roslyn/issues/45728

### CODE REVIEW

**Comment by tmat** [comment]
@Cosifne

**Comment by CyrusNajmabadi** [comment]
What is SBD?

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
```
Comment (mavasani) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs:7]: Should this be outside the Api namespace?
```suggestion
namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting
```

Comment (tmat) [reply src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs:7]: No, this is intentionally an API that UnitTesting will consume.

**Comment by tmat** [comment]
@CyrusNajmabadi SBD = Source Based Discovery

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs:7]: I thought we only intended to expose interfaces to partners, not concrete types. Did that design decision change?

Comment (tmat) [reply src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs:7]: I am not aware of any such rule.

Comment (tmat) [reply src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs:7]: Not sure why would that be needed.

**Comment by Cosifne** [comment]
> 
> 
> What is SBD?

@CyrusNajmabadi  It is a service (own by test team) that used to find unit test from the source file without building the project.

**Comment by tmat** [comment]
Yes, verified.

**Pull request merged**